src/App.js

```diff
@@ -1,26 +1,97 @@
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+      <div className="header">
+        <Header />
+      <section className="options">
+            {/* lets move this to another component?  */}
+
+            <select className="creature-type-filter" onChange={this.handleChange}>
+              <option value="" defaultValue>
+                All Types
+              </option>
+              <option value="narwhal">Narwhal</option>
+              <option value="rhino">Rhino</option>
+              <option value="unicorn">Unicorn</option>
+              <option value="unilego">Unilego</option>
+              <option value="triceratops">Triceratops</option>
+              <option value="markhor">Markhor</option>
+              <option value="mouflon">Mouflon</option>
+              <option value="addax">Addax</option>
+              <option value="chameleon">Chameleon</option>
+              <option value="lizard">Lizard</option>
+              <option value="dragon">Dragon</option>
+            </select>
+
+          </section>
+
+          <section className="horn-options">
+
+            <select className="creature-horn-filter" onChange={this.handleChange}>
+              <option value="" defaultValue>
+                Number of Horns
+              </option>
+              <option value="1">One</option>
+              <option value="2">Two</option>
+              <option value="3">Three</option>
+              <option value="100">One Hundred</option>
+              </select>
+
+          </section>
 
-export default App;
+        
+            <ul className="creatures">
+            
+           { creatureData
+              .filter(creature => {
+                // if there is nothing selected, show ALL CREATURES
+                  if (!this.state.selected) return true;
+          
+                // otherwise only show the creature if the creature keyword is the same as the selected creature keyword
+                  return creature.keyword === this.state.selected;
+                 })
+           
+            .map(animal => {
+
+    return <ImageItem creature={animal}/>
+    })}
+           </ul>
+
+
+           <ul className="horned-creatures">
+            
+           { creatureData
+              .filter(creature => {
+                // if there is nothing selected, show ALL CREATURES
+                  if (!this.state.selected) return true;
+          
+                // otherwise only show the creature if the creature keyword is the same as the selected creature keyword
+                  return creature.horns === this.state.selected;
```

Comment:
"I spent a long time trying to get the Horns filter to work and it only worked when I changed all of the integers in the data.js file to strings. I'd like to know how I could have solved it without doing that."

The problem here is that json returns a string even if it looks like a number. Instead of changing your data type to string, you can change your json return to type number:
return creature.horns === Number(this.state.selected);